Set TargetRubyVersion: 2.0 in .rubocop.yml to avoid unwanted autocorrect with Ruby 2.0+

nafu opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
<!--- Please describe in detail how you tested your changes. --->
There are some rubocop errors with Ruby 2.4.x.
If we run `bundle exec rubocop -a` with Ruby 2.4.x, it automatically corrects some errors.
But we don't want it, fastlane supports 2.0+.

### Description
<!--- Describe your changes in detail -->
Update rubocop configuration and avoid unwanted autocorrect by running `bundle exec rubocop -a`.

nafu committed:
Run bundle exec rubocop -a

nafu committed:
Set TargetRubyVersion of rubocop to 2.4

Currently rubocop use 2.1 parser on CircleCI which is causing parse
errors.
This will fix those errors.
Ref. https://circleci.com/gh/fastlane/fastlane/10316

giginet commented on .rubocop.yml:
```diff
@@ -217,6 +217,7 @@ Layout/SpaceAroundOperators:
     - '**/spec/actions_specs/xcodebuild_spec.rb'
 
 AllCops:
+  TargetRubyVersion: 2.4
```
Is this true?

We are still supporting System Ruby (2.0.0).
I think this setting isn't suitable.

nafu replied:
@giginet 

Thanks. You're right 👍 
Because default ruby version is 2.4.x in my development env, the PR instruction cause autocorrection by rubocop with ruby 2.4.x.

> bundle exec rubocop -a to ensure the code style is valid

Should we define `TargetRubyVersion: 2.0` in .rubocop.yml?

giginet replied:
According to the official document, it should be set minimum support version.

https://github.com/bbatsov/rubocop/blob/master/manual/configuration.md#setting-the-target-ruby-version


```
If .ruby-version exists in the directory RuboCop is invoked in, RuboCop will use the version specified by it. Otherwise, users may let RuboCop know the oldest version of Ruby which your project supports with:
```

So we should set this to `2.0`. 👍

nafu replied:
@giginet 

👍 

I updated branch and issue details!

nafu committed:
Revert "Run bundle exec rubocop -a"

This reverts commit 67a83cbbdbe5f68aaca6573a010c0c00cace9f30.

nafu committed:
Set TargetRubyVersion of rubocop to 2.0

We support ruby 2.0+ and we'd like to avoid unwanted autocorrection
by running `bundle exec rubocop -a`,

See more about the configuration of rubocop:
https://github.com/bbatsov/rubocop/blob/master/manual/configuration.md#setting-the-target-ruby-version

Thanks @giginet
https://github.com/fastlane/fastlane/pull/9819#discussion_r128737202

KrauseFx reviewed (approved):
Nice catch, great change, thanks @nafu ❤️

nafu commented:
Thanks for your review 👍  @giginet @KrauseFx
